Documantation for VSCode extension and dev setup (#1130)
demisto/content-docs: docs/concepts/vscode-extension.md

* You can use this feature together with the content Dev Container.

## Open integrations and scripts in Dev Container (Advanced)

The `XSOAR` integrations and scripts actually run on a dedicated Docker image.
In some integrations, the Python environment is not enough, and it is not easy to install the system dependencies.

With this feature, you are able to develop inside the integration or a script Docker image, with **all** dependencies preinstalled.
This will use the Docker image that is in the `YAML` file.

### System requirements

**Windows**: Docker Desktop 2.0+ on Windows 10 Pro/Enterprise. Windows 10 Home (2004+) requires Docker Desktop 2.3+ and the WSL 2 back-end.
**macOS**: Docker Desktop 2.0+.
**Linux**: Docker CE/EE 18.06+ and Docker Compose 1.21+.

### Installation

Follow the [VSCode documentation](https://code.visualstudio.com/docs/remote/containers#_installation).

### Usage

* Go to the integration or a script.
* Right-click it and select **Open integration/script in a Dev Container**

### Features

The same features of the [virtual environment](#open-integrations-and-scripts-in-python-virtual-environment) feature is available here. 

## Troubleshooting

> - If **Open integration/script in a Dev Container** or **Open integration/script in a virtual environment** fails, make sure the **Docker** is running. In addition, you can try to [clean up the Docker](https://docs.docker.com/config/pruning/) or [sign in to docker](https://www.docker.com/blog/seamless-sign-in-with-docker-desktop-4-4-2/) to avoid the [docker pull rate limit](https://docs.docker.com/docker-hub/download-rate-limit/#:~:text=Pull%20rates%20limits%20are%20based,to%205000%20pulls%20per%20day.).

Reviewer: ```suggestion
### I**Open integration/script in a Dev Container** or **Open integration/script in a virtual environment** fails
1. Make sure **Docker** is running.
2. If *Docker* is running try to [clean up the Docker](https://docs.docker.com/config/pruning/) or [sign in to docker](https://www.docker.com/blog/seamless-sign-in-with-docker-desktop-4-4-2/) to avoid the [docker pull rate limit](https://docs.docker.com/docker-hub/download-rate-limit/#:~:text=Pull%20rates%20limits%20are%20based,to%205000%20pulls%20per%20day.).
```

> - If **demisto-sdk** is not recognized from the extension using **MacOS** and **zsh** terminal, try adding the following to [VSCode settings.json](https://code.visualstudio.com/docs/getstarted/settings#_settingsjson):

Reviewer: ```suggestion
- ### **demisto-sdk** is not available when using **MacOS** and **zsh** terminal.
* try adding the following to [VSCode settings.json](https://code.visualstudio.com/docs/getstarted/settings#_settingsjson):
```
